instance/tests/models/test_openedx_instance.py

```diff
@@ -1251,4 +1253,45 @@ def test_active_vm_dns_records(self, mocks, mock_consul):
         self._verify_vm_dns_records('test.com', [appserver1.server.public_ip])
         appserver2.make_active()
         self._verify_vm_dns_records('test.com', [appserver1.server.public_ip, appserver2.server.public_ip])
-        # We haven't implemented cleaning up VM DNS records yet, so we can't test it.
+
+    @patch('instance.models.mixins.load_balanced.LoadBalancedInstance.remove_dns_records')
+    @patch('instance.models.mixins.openedx_monitoring.OpenEdXMonitoringMixin.disable_monitoring')
+    @patch('instance.models.load_balancer.LoadBalancingServer.reconfigure')
+    @patch_services
+    def test_clean_up_dns_records(self, mocks, *mock):
+        """
+        Test that the DNS records for app servers get removed
+        """
+        # spawn and activate first appserver
+        instance = OpenEdXInstanceFactory(internal_lms_domain='vm-dns.test.com')
+        instance.spawn_appserver()
+        appserver1 = instance.appserver_set.first()
+        appserver1.make_active()
+
+        # spawn and activate second appserver
+        mocks.os_server_manager.set_os_server_attributes('server2', _loaded=True, status='ACTIVE')
+        instance.spawn_appserver()
+        appserver2 = instance.appserver_set.first()
+        appserver2.make_active()
+        self._verify_vm_dns_records('test.com', [appserver1.server.public_ip, appserver2.server.public_ip])
+
+        # spawn and activate third appserver
+        mocks.os_server_manager.set_os_server_attributes('server3', _loaded=True, status='ACTIVE')
+        instance.spawn_appserver()
+        appserver3 = instance.appserver_set.first()
+        appserver3.make_active()
+
+        # spawn and activate fourth appserver
+        mocks.os_server_manager.set_os_server_attributes('server4', _loaded=True, status='ACTIVE')
+        instance.spawn_appserver()
+        appserver4 = instance.appserver_set.first()
+        appserver4.make_active()
+
+        # deactivate first two appservers
+        appserver1.make_active(active=False)
+        appserver2.make_active(active=False)
+        self._verify_vm_dns_records('test.com', [appserver3.server.public_ip, appserver4.server.public_ip])
+
+        # archive instance and verify no dns records
+        instance.archive()
```

Comment:
@pkulkark, nit: testing the removal when archiving an instance should be a separate test.

Reply:
I figured since archive calls the `clean_up_appserver_dns_records` method, this would be a good place to add it. Does that make sense?